app/controllers/CategoryGuidanceController.scala

```diff
@@ -46,20 +47,22 @@ class CategoryGuidanceController @Inject() (
   categorisationService: CategorisationService,
   navigator: Navigator,
   goodsRecordConnector: GoodsRecordConnector,
-  sessionRepository: SessionRepository
+  traderProfileConnector: TraderProfileConnector
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController
     with I18nSupport
     with Logging {
 
-  def onPageLoad(recordId: String): Action[AnyContent] = (identify andThen getData andThen requireData).async {
-    implicit request =>
+  def onPageLoad(mode: Mode, recordId: String): Action[AnyContent] =
+    (identify andThen getData andThen requireData).async { implicit request =>
       categorisationService
         .requireCategorisation(request, recordId)
         .flatMap { userAnswers =>
           val recordCategorisations = userAnswers.get(RecordCategorisationsQuery)
           val categorisationInfo    = recordCategorisations.flatMap(_.records.get(recordId))
           val scenario              = categorisationInfo.map(Scenario.getRedirectScenarios)
+          val areWeRecategorising   = userAnswers.get(RecategorisingQuery(recordId)).getOrElse(false)
```

Comment:
should we be defaulting here? is it not an invalid scenario if we don't have that query

Reply:
It only gets set if we go to the longer commodity code page, so if you are just normally categorising it won't be set.